src/biokbase/narrative/jobs/job.py

```diff
@@ -444,12 +518,30 @@ def _repr_javascript_(self):
             output_widget_info=json.dumps(output_widget_info),
         )
 
+    @staticmethod
+    def _verify_job_id(job_id: str) -> None:
+        if job_id is None:
+            raise ValueError("Cannot create a job without a job ID!")
+
+    @staticmethod
+    def _verify_children(ee2_state, children: List["Job"]) -> None:
+        if children is None or len(children) == 0:
+            raise ValueError(
+                "Job with `batch_job=True` must be instantiated with child job instances"
+            )
+        state_child_ids = ee2_state.get("child_jobs", [])
+        inst_child_ids = [job.job_id for job in children]
+        if sorted(state_child_ids) != sorted(inst_child_ids):
+            raise ValueError(
+                "Child job id mismatch"
+            )
```

Comment:
I think this would be better in the job manager -- that way, if there are missing child jobs, the job manager can fetch them. As it is, there isn't really a way for the job manager or the job object to resolve this situation.